File: Lab2/OOPLab2/OOPLab2/Class/Smartfon.cs
namespace OOPLab2.Class
{
    public class Smartfon : Techinc
    {
        private int cameraMP;
        private bool isCall = false;
        private string manufactures;

        public Smartfon() : base()
        {
            this.cameraMP = 1;
            this.isCall = false;
            this.manufactures = "Huawei";
        }
        public Smartfon(string name, int id, int cameraMP) : base(name, id)
        {
            this.cameraMP = cameraMP;
            this.isCall = false;
            this.manufactures = "Huawei";
        }
        public Smartfon(string name, int id, string country, bool enabled, int cameraMP, bool isCall, string manufactures) 
            : base(name, id, country, enabled)
        {
            this.cameraMP = cameraMP;
            this.isCall = isCall;
            this.manufactures = manufactures;
        }
        public int getCameraMP() { return cameraMP; }
        public string getManufactures() { return manufactures; }
        public bool getIsCall() { return isCall; }

        public void setCameraMP(int cameraMP) { this.cameraMP = cameraMP; }
        public void setManufactures(string manufactures) { this.manufactures = manufactures; }
        public void call() { this.isCall = true; }
        public void reset() { this.isCall = false; }
        public void takePhoto()
        {
            Console.WriteLine($"Сделано фото качеством - ", cameraMP, " мегапикселей");
        }
        public override void displayInfo()
        {
            base.displayInfo();
            Console.WriteLine($"Производитель телефона: ", manufactures);
            Console.WriteLine($"Мегапиксели телефона: ", cameraMP);
        }
        public override double calculatePowerConsumption()
        {
            return enabled ? 5 : 0.5;
        }
    }
}

File: Lab3/labainator3000/src/main/java/org/example/Class/Smartfon.java
package org.example.Class;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import java.io.Serializable;
import java.util.Map;
import java.util.HashMap;

@Entity
@Table(name = "smartfon")
@PrimaryKeyJoinColumn(name = "technic_id")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@ToString(callSuper = true)
public class Smartfon extends Technic implements Serializable {

    @Column(name = "camera_mp")
    private int cameraMP;

    @Column(name = "is_call")
    private boolean isCall = false;

    @Column(name = "manufactures", length = 100)
    private String manufactures;

    public Smartfon(String name, int cameraMP, String manufactures, ComputerShop shop) {
        super(name, shop);
        this.cameraMP = cameraMP;
        this.isCall = false;
        this.manufactures = manufactures;
        this.setCountry("Korea");
        this.setEnabled(false);
    }

    public Smartfon(String name, String country, boolean enabled, int cameraMP, boolean isCall, String manufactures, ComputerShop shop) {
        super(name, country, enabled, shop);
        this.cameraMP = cameraMP;
        this.isCall = isCall;
        this.manufactures = manufactures;
    }

    // Бизнес-методы
    public void takePhoto() {
        System.out.println("Сделано фото качеством " + cameraMP + " мегапикселей");
    }

    public void call() {
        this.isCall = true;
    }

    public void reset() {
        this.isCall = false;
    }

    @Override
    public Map<String, Object> displayInfo() {
        Map<String, Object> result = super.displayInfo();
        result.put("cameraMP", cameraMP);
        result.put("isCall", isCall ? "звонит" : "не звонит");
        result.put("manufacture", manufactures);
        result.put("type", "Smartfon");
        return result;
    }

    @Override
    public double calculatePowerConsumption() {
        return isEnabled() ? 5 : 0.5;
    }
}